docs/guides/db-connection-pooling.md

```diff
@@ -0,0 +1,151 @@
+---
+id: db-connection-pooling
+title: Database Connection Pooling
+sidebar_label: Database Connection Pooling
+---
+
+Opening a database connection is an expensive process, and that's why we use **[Connection pool](https://en.wikipedia.org/wiki/Connection_pool)** to reduce the cost. And it also allows us to have a good transaction management in a single session that uses different providers.
+
+The example uses PostgreSQL and **[node-postgres](https://node-postgres.com/features/transactions)** below;
+
+We define two providers in `DatabaseModule`, the first one is `Pool` which will be application scoped, and `DatabaseProvider` will be session-scoped. So, it will provide us different clients from connection pool for each session/network request. See Dependency Injection part of our docs to learn more about provider scopes.
+`database.module.ts`
+```ts
+import { Pool } from 'pg';
+export const DatabaseModule = new GraphQLModule({
+    providers: [
+        Pool,
+        DatabaseProvider
+    ]
+});
+```
+
+And we will use `OnResponse` hook to release the client to the pool after we've done with it. See Dependency Injection part of our docs to learn more about hooks.
+`database.provider.ts`
+```ts
+import { Injectable } from '@graphql-modules/di';
+import { SQLStatement } from 'sql-template-strings';
+import { Pool, PoolClient } from 'pg';
+
+@Injectable({
+    scope: ProviderScope.Session
+})
+export class DatabaseProvider implements OnResponse {
+    private _poolClient: PoolClient;
+    constructor(private pool: Pool) {}
```

Comment:
Mention here that the `DatabaseProvider` is created on a session level, but the instance of `Pool` will be the same instance always